<launch>

	<!-- formulas: -->
	<rosparam command="load" file="$(find hybrid_controller)/launch/formulas.yaml" />
	<rosparam command="load" file="$(find hybrid_controller)/launch/formula_derivatives.yaml" />


	<rosparam param="u_max">[10.0, 6.28]</rosparam>
	<param name="K" type="int" value="10"/>
	<param name="n_robots" type="int" value="2"/>


	<!-- controllers: -->
	<node pkg="hybrid_controller" type="hybrid_controller_node" name="hybrid_controller0" output="screen">
		<remap from="/cmdvel" to="/cmdvel0" />
		<remap from="/c" to="/c0" />
		<remap from="/controller_params" to="/controller_params0" />
		<param name="~robot_id" type="int" value="0"/>
	</node>
	<node pkg="hybrid_controller" type="hybrid_controller_node" name="hybrid_controller1" output="screen">
		<remap from="/cmdvel" to="/cmdvel1" />
		<remap from="/c" to="/c1" />
		<remap from="/controller_params" to="/controller_params1" />
		<param name="~robot_id" type="int" value="1"/>
	</node>
	<!--
	<node pkg="hybrid_controller" type="hybrid_controller_node" name="hybrid_controller2" output="screen">
		<remap from="/control_input" to="/control_input_robot2" />
        <remap from="/pose" to ="/pose_robot2" />
	</node>
	-->


	<!-- virtual robots -->
	<node pkg="robot" type="robot_node" name="robot0" output="screen">
		<remap from="/control_input" to="/cmdvel0" />
		<remap from="/pose" to="/pose_robot0" />
	</node>
	<node pkg="robot" type="robot_node" name="robot1" output="screen" >
		<remap from="/control_input" to="/cmdvel1" />
		<remap from="/pose" to="/pose_robot1" />
	</node>
	<!--
	<node pkg="robot" type="robot_node" name="robot2" output="screen" >
		<remap from="/control_input" to="/control_input_robot2" />
		<remap from="/pose" to="/pose_robot2" />
	</node>
	-->


	<node pkg="rviz" type="rviz" name="rviz" />
	
</launch>
